compiler/src/dotty/tools/MainGenericRunner.scala

```diff
@@ -157,7 +157,7 @@ object MainGenericRunner {
         process(tail, newSettings.withResidualArgs(arg))
 
   def main(args: Array[String]): Unit =
-    val scalaOpts = envOrNone("SCALA_OPTS").toArray.flatMap(_.split(" "))
+    val scalaOpts = envOrNone("SCALA_OPTS").toArray.flatMap(_.split(" ")).filter(_.nonEmpty)
```

Comment:
This is handling empty value of env variable, without it, we were passing empty String